sdk/tests/deqp/modules/shared/glsShaderExecUtil.js

```diff
@@ -446,7 +446,11 @@ goog.scope(function() {
             else if (gluShaderUtil.isDataTypeMatrix(basicType)) {
                 /** @type {number} */ var numRows = gluShaderUtil.getDataTypeMatrixNumRows(basicType);
                 /** @type {number} */ var numCols = gluShaderUtil.getDataTypeMatrixNumColumns(basicType);
-                /** @type {number} */ var stride = numRows * numCols * 4;//sizeof(float);
+                // A matrix consists of several (column-major) vectors. A buffer is created for
+                // every vector in gluDrawUtil.draw() below. Data in every buffer will be tightly
+                // packed. So the stride should be 0. This is different from the code in native
+                // deqp, which use only one buffer for a matrix, the data is interleaved.
+                /** @type {number} */ var stride = 0;
```

Comment:
@kenrussell , I have no idea about the exact workload of porting the latest C++ deqp code to JS. But when we debug the errors introduced by porting, I think it is really a big effort. You know, there are tremendous differences between C++ and JS, and some subtle differences between GLES and webgl, and so on. All of these will lead to lots of redesign and/or workarounds. I can not say they are very difficult, but it is definitely time-consuming. 

@gyagp want to try to use some tool (emscripten or something like emscripten) to auto-generate the JS code from C++, then manually rewrite the auto-porting code and refine the patterns of the tool. But he has no time currently. You know, in this way, we can reuse the tool and make the tool better and better every time we port the code from C++ to JS when WebGL is upgraded. However, I am not sure whether this method can save some time. WDYT?